runelite-client/src/main/java/net/runelite/client/Notifier.java

```diff
@@ -293,4 +330,46 @@ private static String toUrgency(TrayIcon.MessageType type)
 				return "normal";
 		}
 	}
+
+	private void playCustomSound()
+	{
+		Clip clip = null;
+
+		// Try to load the user sound from ~/.runelite/notification.wav
+		File file = new File(RuneLite.RUNELITE_DIR, "notification.wav");
+		if (file.exists())
+		{
+			try
+			{
+				InputStream fileStream = new BufferedInputStream(new FileInputStream(file));
+				try (AudioInputStream sound = AudioSystem.getAudioInputStream(fileStream))
+				{
+					clip = AudioSystem.getClip();
+					clip.open(sound);
+				}
+			}
+			catch (UnsupportedAudioFileException | IOException | LineUnavailableException e)
+			{
+				clip = null;
+				log.warn("Unable to play notification sound", e);
+			}
+		}
+
+		if (clip == null)
+		{
+			// Otherwise load from the classpath
+			InputStream fileStream = new BufferedInputStream(Notifier.class.getResourceAsStream("notification.wav"));
+			try (AudioInputStream sound = AudioSystem.getAudioInputStream(fileStream))
+			{
+				clip = AudioSystem.getClip();
+				clip.open(sound);
+			}
+			catch (UnsupportedAudioFileException | IOException | LineUnavailableException e)
+			{
+				log.warn("Unable to play builtin notification sound", e);
+				return;
+			}
+		}
+		clip.start();
```

Comment:
This looks like it would fail if the built-in notification fails, maybe do same this as with external one and set it to null and add null-check. And in case it fails completely maybe revert back to System.beep() at least.